sumpy/p2p.py

```diff
@@ -473,33 +476,49 @@ def get_kernel(self, max_nsources_in_one_box, max_ntargets_in_one_box,
             "{[iknl]: 0 <= iknl < noutputs}",
             "{[isrc_box]: isrc_box_start <= isrc_box < isrc_box_end}",
             "{[idim]: 0 <= idim < dim}",
-            "{[isrc]: isrc_start <= isrc < isrc_end}"
         ]
 
-        src_outer_limit = (max_nsources_in_one_box - 1) // nsplit
-        tgt_outer_limit = (max_ntargets_in_one_box - 1) // nsplit
-
-        if gpu:
+        # This is the amount of targets we process per thread
+        ntgt_offset = (max_ntargets_in_one_box - 1) // work_items_per_group + 1
+        # This is the amount of sources we pre-fetch per thread
+        nsrc_offset_prefetch = (max_nsources_in_one_box - 1) \
+                // work_items_per_group + 1
+        # This is the amount of sources we process per pre-fetch
+        nsrc_offset = (max_nsources_in_one_box - 1) // nprefetch + 1
+        # This is the amount of sources we pre-fetch per thread per pre-fetch
+        nsources_prefetch_per_thread = (nsrc_offset_prefetch - 1) // nprefetch + 1
```

Comment:
IMO, this type of thing is exactly what `split_iname` is for. Did that not work?

Reply:
I spent yesterday trying to and the least horrible way to do this with transformations was with https://github.com/isuruf/sumpy/pull/new/less_local2
Which do you prefer?